pkg/collector/scheduler/scheduler_test.go

```diff
@@ -126,6 +132,23 @@ func TestStop(t *testing.T) {
 	assert.Nil(t, s.Stop())
 }
 
+func TestStopCancelsProducers(t *testing.T) {
+	s := getScheduler()
+	minAllowedInterval = time.Millisecond // for the purpose of this test, so that the scheduler actually schedules the checks
+	defer resetMinAllowedInterval()
+
+	s.Enter(&TestCheck{intl: time.Millisecond})
+	s.Run()
+
+	time.Sleep(2 * time.Millisecond) // wait for the scheduler to actually schedule the check
+
+	s.Stop()
+	// once the scheduler is stopped, it should be safe to close this channel. Otherwise, this should panic
+	close(s.checksPipe)
+	// sleep to make the runtime schedule the hanging producer goroutines, if there are any left
+	time.Sleep(time.Millisecond)
```

Comment:
any way we could assert the hanging producer goroutines were indeed scheduled? Otherwise, why sleep at all? Safety?

Reply:
I'm not aware of and haven't found any way to check that the go scheduler did schedule the other hanging goroutines.

Here the test wants to make sure that the other goroutines don't cause panics once the `s.checksPipe` channel is closed. Locally I've been able to consistently make the test cause panics with such a sleep. Whereas without a sleep the hanging goroutines often are never scheduled and the tests pass with the faulty code.

So overall this seems like a reasonable way to force the hanging goroutines to be scheduled